detailed-design/threat-model.md

```diff
@@ -1,41 +1,100 @@
 
-### Read Level Attacks
-There is a spectrum of severity. The least severe is a side-channel where the attacker can find some information about a specific transaction. The most severe is when the attacker can extract the encryption key and can read all transactions.
+#### Read-Level Attacks
+There is a spectrum of attack severity. The least severe is a side-channel where the attacker can find some information about a specific transaction. The most severe is when the attacker can extract the master secret and can read all present and future transactions.
 
-If this attacker is discrete, the information leak can continue until a software patch is published or until new hardware that removes this attack is released.
+If this attacker is discreet, the information leak can continue until a software patch is published or until new hardware that removes this attack is released.
 
-This threat is specific to Obscuro. If the attack is successful, the protocol reverts to the behavior of a typical public blockchain where all transactions are public, and MEV is possible.
+This threat is specific to confidential blockchains. If the attack is successful, the protocol reverts to the behavior of a typical public blockchain where all transactions are public, and MEV is possible.
 
-### Write Level Attacks
+#### Write-Level Attacks
 There are a few variants of this attack.
 
-The most severe is when the attacker impersonates a TEE and signs a rollup containing illegal transactions. If there is no withdrawal attempt in the rollup, this attack is equivalent to an L1 node publishing an invalid block. All the honest TEEs will ignore it. Even in this case, the protocol reverts to the behavior of a typical blockchain.
+The most severe is when the attacker impersonates a TEE and signs a rollup containing illegal transactions. If there is no withdrawal attempt in the rollup, or there are no other external side effects, this attack is equivalent to an L1 node publishing an invalid block. All the honest TEEs will ignore it. Even in this case, the protocol reverts to the behavior of a typical blockchain.
+
+If the attacker impersonating an enclave attempts to withdraw funds from the bridge contract based on invalid transactions, is a case that needs special consideration. This is where the L1 protocol meets the L2 protocol, and the bridge contract has to decide the validity of the withdrawal request in order to execute it. The solution is described in detail in the POBI protocol.
+
+#### Colluding Write Level Attacks
+This is an extreme attack on this protocol, where all aggregators find a way to break the TEE and are colluding to steal all funds.
+
+The defense is to have a reasonable number of active verifiers that have a stake in the good functioning of Obscuro. "Active" means that they are watching the Obscuro ledger in real time. These actors will detect if there is a malicious head rollup and no other valid fork is being aggregated.
```

Comment:
Not sure this needs to be added to the WP but we should not assume enough people / verifiers will actively watch for good behaviour. Most either won't care enough or will assume someone else is watching.

Reply:
That's a good point. But there will be someone watching, and that is the governance of Obscuro.
I hinted at that with: "that have a stake in the good functioning of Obscuro."
Should it be more explicit?